### PR TITLE
Add new endpoint to assess whether a new password is acceptable

### PR DESCRIPTION
**Pull request by Schparky**
This new endpoint (`PUT /user/{id}/password/assess`) does the same checks as for updating a password (`PUT /user/{id}/password`) but does not save the new password.

### CODE REVIEW

File: api.raml [hunk 1]
```diff
@@ -443,17 +443,6 @@ types:
               type: UserResponse
           204:
             description: No user record was found with that `employee_id`.
-          409:
```
Comment (wcjr) [inline api.raml]: I'm surprised to see this removed...am I missing something?

File: api.raml [hunk 2]
```diff
@@ -464,6 +453,34 @@ types:
...
+              }
+          responses:
+            204:
+              description: >
+                The `password` is valid for the user with the given `employee_id`.
+            404:
+              description: No user found with given `employee_id`.
+              body:
+                type: Error
+            422:
```
Comment (wcjr) [inline api.raml]: `409`?

File: api.raml (hunk 1 above)
Comment (Schparky) [reply api.raml]: My mistake. At one point I had considered changing it to 422 like all other model validations. I'll correct this.

File: application/common/models/User.php [hunk 3]
```diff
@@ -914,4 +914,19 @@ public function afterFind()
...
+
+    public function assessPassword(string $newPassword): bool
+    {
+        $password = new Password();
+
+        $password->user_id = $this->id;
+        $password->password = $newPassword;
+
+        if (! $password->validate()) {
+            $this->addErrors($password->getErrors());
```
Comment (forevermatt) [inline application/common/models/User.php]: Are the errors returned by `getErrors()` associated with specific attribute names? I didn't know if taking a `Password` record's errors and setting them on a `User` record would have trouble due to there being different attribute names.

Comment (Schparky) [reply application/common/models/User.php]: Seems to work. Also, I just took a glance at the Yii code to confirm it doesn't check the value of the array key.

File: api.raml (hunk 2 above)
Comment (Schparky) [reply api.raml]: Changed from 422 to 409.

File: application/common/models/Password.php [hunk 4]
```diff
@@ -194,17 +197,13 @@ public function getUser()
...
-     * @return bool
+     * @param $attribute
      * @throws ConflictHttpException
      */
-    public function save($runValidation = true, $attributeNames = null)
+    public function checkRecentlyUsed($attribute)
     {
         if ($this->hasAlreadyBeenUsedTooRecently()) {
-            throw new ConflictHttpException('May not be reused yet.', 1542395933);
+            throw new ConflictHttpException('May not be reused yet', 1542395933);
```
Comment (fillup) [inline application/common/models/Password.php]:
```suggestion
            throw new ConflictHttpException('You have already used this password, please use a new password', 1542395933);
```

Comment (Schparky) [reply application/common/models/Password.php]: This message is swallowed by `idp-pw-api`, and passes on the 409 to the UI with its own translated message.  In fact, as it stands today, the UI overrides the API message with its own.

**Pull request merged**